add open graph and page titles

itstombellis opened the pull request:
Closes #37

itstombellis committed:
Added General OG To base.tmpl

I have created a general image to be used whenever a more specific one
isn’t provided (such as show image).
Created general og meta tags, and added twitter card.

itstombellis committed:
Removed Comment

I understand the type tag more now.

itstombellis committed:
Created OG Tags For people.tmpl

I have created the idea behind the OG tags, need to check syntax and
answer the comments.

itstombellis committed:
Created OG Tags for getinvolved.tmpl

Created an image to be associated with the get involved page, I have
also added tags. There are some comments on this page that need
addressing.

itstombellis committed:
Updated Naming Convention For Images

For open graph images, I am using naming convention:
open_graph-page_title

itstombellis committed:
Created OG Tags for show.tmpl

I have added OG Tags for show.tmpl.
There are comments that need resolved.

itstombellis committed:
Tags contained with open-graph block on people.tmpl

Now implemented tags within the open-graph block. Still contains some
TODOs

itstombellis committed:
General Tidy of open-graph block

Re-ordered meta tags.
Added todo, getting URL for page.

itstombellis committed:
Implements open-graph block on get involved.tmpl

Moved the tags to inside the open-graph block.

itstombellis committed:
Implements open-graph block in show.tmpl

I have added the meta tags into the open-graph block of show.tmpl.

There is still a couple of TODOs.
- Remove HTML from show descriptions.
- Get URL for page.

itstombellis committed:
Fixed URL Tag on show.tmpl

Fixed the URL Tag to use the correct URL for type of page.

itstombellis committed:
Fixed URL Tag on getinvolved.tmpl

Fixed the URL Tag to use the correct URL for type of page.

itstombellis committed:
Updated OG Tags in people.tmpl

- Changed the URL to be the url of the person.
- Updated Username so the tag should work (with pr #43).
- Title now uses {{.longName}} instead of University Radio York.
- Now uses attributes first_name, last_name

itstombellis committed:
Title now uses {{.longName}} in people.tmpl

Title now uses {{.longName}} in people.tmpl

itstombellis committed:
Replaced `University Radio York` in titles too `{{.longName}}`

Updated base.tmpl and show.tmpl to use `{{.longName}}` in the OG Title
tags rather than `University Radio York`.

itstombellis committed:
Corrected Block to define

itstombellis committed:
Added Title Content To Each Template

Brookke committed:
Merge branch 'development' into tom-openGraph

Brookke committed:
Merge branch 'development' into tom-openGraph

Brookke committed:
more consistent variable usage

itstombellis committed:
Updated OpenGraph URLs to use URL Prefix

Brookke committed:
Update _buttons.scss

itstombellis committed:
Removed Dots From Definitions of Title and OpenGraph

itstombellis committed:
Merge branch 'tom-openGraph' of https://github.com/UniversityRadioYork/2016-site into tom-openGraph

itstombellis committed:
Use long url

itstombellis committed:
Identified Issues

itstombellis committed:
Fixed Open Graph Errors on show.tmpl Still need to strip html from description.

itstombellis committed:
People.tmpl relies on #45

itstombellis commented:
Need to wait for #45 to be merged. In order to check it works.

itstombellis committed:
Open Graph Requires the full URLs, update all open-graph blocks to use full URL.

itstombellis committed:
Fixed typing error

itstombellis committed:
Merge branch 'development' into tom-openGraph

itstombellis committed:
Implements Changes From #45

itstombellis committed:
Changes from #45

itstombellis committed:
Removed Pesky Whitespace

itstombellis committed:
Added General Image To Seasons

itstombellis committed:
Updated URYNews Twitter Handle

Brookke commented on views/about.tmpl:
```diff
@@ -1,3 +1,17 @@
+
+{{define "open-graph"}}
+    <meta name="twitter:card" content="summary_large_image"/>
+    <meta name="twitter:site" content="@URY1350"/>
```
Tempted to say we should add the twitter handle to the config file as will will want it for the twitter links etc @mstratford @LordAro what do you guys think?

itstombellis replied:
Yeah that sounds good.
Open-graph requires the full URL (https://ury.org.uk/...) is it worth putting this in the config also?

Brookke commented on views/about.tmpl:
```diff
@@ -1,3 +1,17 @@
+    <meta property="og:type" content="website"/>
+    <meta property="og:locale" content="en_GB" />
+    <meta property="og:title" content="{{.PageContext.LongName}} | About"/>
+    <meta property="og:description" content="University Radio York (URY) is the student radio station for the University of York, broadcasting 24 hours a day during term time on 1350am across campus, and worldwide via this website."/>
```
Also should we put the site description in the config too

Brookke commented on views/show.tmpl:
```diff
@@ -1,3 +1,24 @@
+{{define "title"}}{{.PageContext.ShortName}} | {{with .PageData}}{{.Show.Title}}{{end}}{{end}}
+
+{{define "open-graph"}}
+	<meta name="twitter:card" content="summary_large_image"/>
```
Shows might suit the [summary type](https://dev.twitter.com/cards/types/summary) rather than [large summary](https://dev.twitter.com/cards/types/summary-large-image).

itstombellis replied:
@Brookke yeah that sounds a good suggestion. I think I will do that to people too?

Brookke commented:
Overall looks good, just a few comments I have added

itstombellis committed:
Added FullURL to config, this means that "https://ury.org.uk" doesn't have to be typed also making testing easier.

itstombellis committed:
Added social media to config

itstombellis committed:
Added Site Description to config

itstombellis committed:
Changed People, Season, Show to use Twitter Summary.

itstombellis commented:
@Brookke I have made the changes you suggested as I think they made sense. (If they don't I can undo the changes).

Brookke commented:
Wondering if we should extend @MattWindsor91's URL prefixing or something for the fullURL, not sure what is best

mstratford commented:
Yeah, if the open graph stuff could auto set the subdirectory using the above, that'd be great

itstombellis committed:
Corrected Spelling on Homepage

itstombellis committed:
Added Pinned Tab Icon

Brookke commented:
If you can use the url prefixer on the urls you have added so far, we can look into maybe extending it to work with the full url at a later date

itstombellis committed:
Merge branch 'development' into tom-openGraph

itstombellis committed:
Uses URL-prefixer

Brookke commented on views/show.tmpl:
```diff
@@ -7,12 +7,12 @@
 	<meta property="og:title" content="{{.PageContext.ShortName}} | {{.PageData.Show.Title}}"/>
 	{{if .PageData.Show.Photo}}
-		<meta property="og:image" content="{{.PageContext.FullURL}}{{.PageData.Show.Photo}}"/>
+		<meta property="og:image" content="{{.PageContext.FullURL}}{{url ""}}{{.PageData.Show.Photo}}"/>
```
{{url .PageData.Show.Photo}} should work here

itstombellis committed:
Implemented Changes

Brookke commented on views/season.tmpl:
```diff
@@ -1,3 +1,15 @@
+
+{{define "open-graph"}}
+	<meta name="twitter:card" content="summary"/>
+	<meta name="twitter:site" content="@{{.PageContext.MainTwitter}}"/>
```
The indentation here is too much not sure any indentation is needed on any of the open graph blocks

itstombellis committed:
Remove Indentation

Brookke committed:
removed /> from meta tags

as per [html5
spec](https://www.w3.org/TR/html5/document-metadata.html#the-meta-elemen
t)

Brookke commented:
This is ready as far as I am concerned however, looking at the icon you have added to pinned tabs, it looks very small and the URY is nearly impossible to read.
![screenshot 2017-06-10 23 12 09](https://user-images.githubusercontent.com/10200358/27006686-8407d5f8-4e32-11e7-8c75-caef9ea3226a.png)
(github logo there for comparison) 
I would suggest something maybe like this 
![screenshot 2017-06-10 23 26 14](https://user-images.githubusercontent.com/10200358/27006745-4060923e-4e34-11e7-93d8-a7530597658f.png) or 
![screenshot 2017-06-10 23 29 34](https://user-images.githubusercontent.com/10200358/27006761-b262639e-4e34-11e7-8cad-e5c91c9564d3.png)

Brookke commented:
URY without the (()) isnt the worst either 
![screenshot 2017-06-10 23 31 21](https://user-images.githubusercontent.com/10200358/27006767-f6fc265c-4e34-11e7-819d-c704085fd4a4.png)

itstombellis commented:
@Brookke I quite like the option without brackets. Do you have the SVG for that?

Brookke commented:
I do, but before we continue, its worth noting that we should probably use the same for the Fav ICON too and I am concerned about using too much text in such a small image. This is another option
![screenshot 2017-06-10 23 42 19](https://user-images.githubusercontent.com/10200358/27006804-7e02466c-4e36-11e7-9bc6-a7cd720b8ea9.png)

Brookke commented:
For reference the current favicon used on https://ury.org.uk 
![favicon](https://ury.org.uk/favicon.ico)

itstombellis commented:
They are the old brackets, the new ones are not feathered.

Brookke commented:
@TomBellis I know but that's whats currently on ury site

Brookke committed:
add updated pinned tab icon

LordAro commented:
My own personal opinion, i quite like the `((`, although `URY` and `((U` aren't bad either. `((URY))` is definitely too small

Branding issues should probably be put to marketing though...

itstombellis commented:
@Brookke and I were talking about the double brackets. 
For the favicon we were thinking blue background with white brackets. 
![favicon](https://user-images.githubusercontent.com/16823823/27009497-460643f6-4e88-11e7-839b-c8db6ce72449.png)

For the pinned tab icon have blue brackets on a transparent background.
![screenshot 2017-06-11 00 19 30](https://user-images.githubusercontent.com/16823823/27009499-4babf706-4e88-11e7-9ad8-c3d86c579876.png)

Brookke committed:
add viewbox to svg to ensure the icon fits in the tab correctly

Brookke committed:
add favicons

Brookke commented:
Have added the favicon ![favicon](https://raw.githubusercontent.com/UniversityRadioYork/2016-site/a998601a54a3bebfa5283f18034cf80da6b41a8a/public/favicon.png)

however I am still not certain about the lack of padding this one might be better
![screenshot 2017-06-11 17 35 08](https://user-images.githubusercontent.com/10200358/27012683-92c49280-4ecc-11e7-9be4-8998b8a54158.png)

Brookke committed:
remove old pinnedTab image
